src/main/java/com/nvidia/spark/rapids/jni/GpuTimeZoneDB.java

```diff
@@ -55,18 +60,24 @@ public class GpuTimeZoneDB {
   private static Map<String, Integer> zoneIdToTable;
 
   // use this reference to indicate if time zone cache is initialized.
-  private static HostColumnVector fixedTransitions;
+  private static HostColumnVector transitions;
+  // initial year set to 1900 because some transition rules start early
+  private static final int initialTransitionYear = 1900;
+  private static long maxTimestamp;
+  private static final ZoneId utcZoneId = ZoneId.of("UTC");
 
   /**
    * This should be called on startup of an executor.
    * Runs in a thread asynchronously.
    * If `shutdown` was called ever, then will not load the cache
    */
-  public static void cacheDatabaseAsync() {
+  public static void cacheDatabaseAsync(int maxYear) {
     // start a new thread to load
     Runnable runnable = () -> {
       try {
-        cacheDatabaseImpl();
+        cacheDatabaseImpl(maxYear);
+        maxTimestamp = LocalDateTime.of(maxYear+1, 1, 1, 0, 0, 0)
```

Comment:
```suggestion
        maxTimestamp = LocalDateTime.of(maxYear-1, 1, 1, 0, 0, 0)
```

Here we use maxYear + 1, but for the transition items, we check:
```scala
for (int year = initialTransitionYear; year <= finalTransitionYear; year++)
```
The initialTransitionYear is maxYear.
In `shouldFallbackToCpu` we actually use  maxYear + 1.
We should use a greater value for transition items, right?

Reply:
There are two converts: `fromTimestampToUtcTimestamp` and `fromUtcTimestampToTimestamp`, and there are two kinds of timezones: plus and nagative, e.g.: +10:00, -10:00. If the `input`  represents a local time and time zone is -10:00, then its UTC timestamp value is original value + 10 hours which will be out of range. Using a greater value for transition items can make sure there is no problem. I did not test, but I guess there might be a problem. Would you please test the corner cases.

Reply:
You mean in the case that the date time is 9999? I suppose I did not account for that edge case properly, I will set it to be maxYear+1,1,2 then, this should cover it.

Reply:
Cosidering the following:
```scala
maxYear = 2200
maxTimestamp = LocalDateTime.of(maxYear+1, 1, 1, 0, 0, 0)
for (int year = initialTransitionYear; year <= finalTransitionYear; year++)
shouldFallbackToCpu(cv)
```

When calling `shouldFallbackToCpu`, the data in `cv` may contain a equivalent timestamp: `2201-01-01 08:00:00`  cosidering of time zone offset, and its literal value is < `2201-01-01`,  but the max transition offset rule is only up to `2201-01-01 00:00:00`